internal/defact/default.go

```diff
@@ -0,0 +1,262 @@
+package defact
+
+import (
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"regexp"
+	rt "runtime"
+	"strings"
+
+	"github.com/ActiveState/cli/internal/config"
+	"github.com/ActiveState/cli/internal/errs"
+	"github.com/ActiveState/cli/internal/fileutils"
+	"github.com/ActiveState/cli/internal/locale"
+	"github.com/ActiveState/cli/internal/logging"
+	"github.com/ActiveState/cli/internal/output"
+	"github.com/ActiveState/cli/internal/primer"
+	"github.com/ActiveState/cli/internal/strutils"
+	"github.com/ActiveState/cli/pkg/platform/runtime"
+	"github.com/gobuffalo/packr"
+	"github.com/thoas/go-funk"
+)
+
+type DefaultConfigurer interface {
+	Set(key string, value interface{})
+	GetString(key string) string
+}
+
+type exeFile struct {
+	fpath string
+	name  string
+	ext   string
+}
+
+func getDefaultProjectPath(d DefaultConfigurer) string {
+	return d.GetString("default_project_path")
+}
+
+func setDefaultProjectPath(d DefaultConfigurer, path string) {
+	d.Set("default_project_path", path)
+}
+
+type primable interface {
+	primer.Outputer
+}
+
+// isShimAndTargetsDir check if the specified filename is (probably) a shim and targets a file in the specified directory
+// This function is used during rollback to clean old shims
+func isShimAndTargetsDir(filename, dir string) bool {
```

Comment:
This took me multiple passes of the function and its contents to understand. The naming and description isn't explaining what's happening here very well.

Let's make this a lot clearer, name it something like

```
isShimFor(shimFile, runtimeDir string) bool
```